prow/cmd/sinker/main.go

```diff
@@ -137,21 +137,28 @@ func main() {
 		logrus.WithError(err).Fatal("Error creating manager")
 	}
 
-	buildClusterClients, err := o.kubernetes.BuildClusterClients(cfg().PodNamespace, o.dryRun.Value)
+	buildManagers, err := o.kubernetes.BuildClusterManagers(o.dryRun.Value,
+		func(o *manager.Options) {
+			o.Namespace = cfg().PodNamespace
+		},
+	)
 	if err != nil {
-		logrus.WithError(err).Fatal("Error creating build cluster clients.")
+		logrus.WithError(err).Error("Failed to construct build cluster managers. Is there a bad entry in the kubeconfig secret?")
 	}
 
-	podClients := map[string]podInterface{}
-	for cluster, client := range buildClusterClients {
-		podClients[cluster] = client
+	buildClusterClients := map[string]ctrlruntimeclient.Client{}
+	for clusterName, buildManager := range buildManagers {
+		if err := mgr.Add(buildManager); err != nil {
+			logrus.WithError(err).Fatal("Failed to add build cluster manager to main manager")
+		}
+		buildClusterClients[clusterName] = buildManager.GetClient()
```

Comment:
Please update the unit tests to validate that we clean pods from all build clusters.

Reply:
We already have unittests that validate that, the bug here is because we didn't insert the correct client, not because we didn't use the correct client. Testing this is pretty difficult and would require us to build a `fakeManager` that is then used in unittests.

Reply:
Fair enough.
/lgtm
/approve